### PR TITLE
[native] Add a sample plan validator and e2e tests.

### PR DESCRIPTION
**Pull request by amitkdutta**
## Description
This PR adds a sample validator in worker that fails if plan has nested loop join. We also address comments around naming as in https://github.com/prestodb/presto/pull/23423#issuecomment-2287717299

## Motivation and Context
Adding a sample validaotr shows how the validation interface can used.

## Impact
None

## Test Plan
E2E tests

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by aditi-pandit** [comment]
@amitkdutta @pedroerp : Cross posting my comment https://github.com/prestodb/presto/pull/23423#issuecomment-2287717299 for a sample plan validator.

File: presto-native-execution/presto_cpp/main/common/Configs.h [hunk 1]
```diff
@@ -639,6 +639,8 @@ class SystemConfig : public ConfigBase {
   static constexpr std::string_view kSinkMaxBufferSize{"sink.max-buffer-size"};
   static constexpr std::string_view kDriverMaxPagePartitioningBufferSize{
       "driver.max-page-partitioning-buffer-size"};
+  static constexpr std::string_view kFailOnNestedLoopJoin{
+      "fail-on-nested-loop-join"};
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/common/Configs.h:643]:
```
kPlanValidatorFailOnNestedLoopJoin
velox-plan-validator-fail-on-nested-loop-join?
```

**Review by xiaoxmeng — COMMENTED** [review]
@amitkdutta LGTM. Thanks!

File: presto-native-execution/presto_cpp/main/types/VeloxPlanValidator.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      return true;
+    }
+  }
+
+  return false;
+}
+
+void VeloxPlanValidator::validatePlanFragment(
+    const velox::core::PlanFragment& fragment) {
+  auto failOnNestedLoopJoin =
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/VeloxPlanValidator.cpp:37]:
```
const auto failOnNestedLoopJoin
```

File: presto-native-execution/presto_cpp/main/types/VeloxPlanValidator.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      SystemConfig::instance()
+          ->optionalProperty<bool>(
+              SystemConfig::kPlanValidatorFailOnNestedLoopJoin)
+          .value_or(false);
+  if (failOnNestedLoopJoin == true) {
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/VeloxPlanValidator.cpp:42]:
```
if (failOnNestedLoopJoin) {
```

File: presto-native-execution/presto_cpp/main/types/VeloxPlanValidator.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    VELOX_CHECK(
+        !planHasNestedJoinLoop(fragment.planNode),
+        "Plan uses nested join loop which isn't supported.");
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/VeloxPlanValidator.cpp:45]:
```
Velox plan uses ...
```

**Review by xiaoxmeng — APPROVED** [review]
@amitkdutta LGTM % nits.

**Pull request merged**